test/widgets/action_sheet_test.dart

```diff
@@ -99,6 +103,302 @@ void main() {
+        return;
+      }
+      check(find.byType(BottomSheet)).findsOne();
+
+      for (final buttonFinder in expectedButtonFinders) {
+        check(buttonFinder).findsOne();
+      }
+      check(find.bySubtype<UserTopicUpdateButton>())
+        .findsExactly(expectedButtonFinders.length);
+    }
+
+    void checkUpdateUserTopicRequest(UserTopicVisibilityPolicy expectedPolicy) async {
+      check(connection.lastRequest).isA<http.Request>()
+        ..url.path.equals('/api/v1/user_topics')
+        ..bodyFields.deepEquals({
+          'stream_id': '${channel.streamId}',
+          'topic': topic,
+          'visibility_policy': '${expectedPolicy.toJson()}',
+        });
+    }
+
+    testWidgets('unmuteInMutedChannel', (tester) async {
+      await setupToTopicActionSheet(tester,
+        isChannelMuted: true,
+        visibilityPolicy: UserTopicVisibilityPolicy.none);
+      await tester.tap(unmuteInMutedChannel);
+      await tester.pump();
+      checkUpdateUserTopicRequest(UserTopicVisibilityPolicy.unmuted);
+    });
+
+    testWidgets('unmute', (tester) async {
+      await setupToTopicActionSheet(tester,
+        isChannelMuted: false,
+        visibilityPolicy: UserTopicVisibilityPolicy.muted);
+      await tester.tap(unmute);
+      await tester.pump();
+      checkUpdateUserTopicRequest(UserTopicVisibilityPolicy.none);
+    });
+
+    testWidgets('mute', (tester) async {
+      await setupToTopicActionSheet(tester,
+        isChannelMuted: false,
+        visibilityPolicy: UserTopicVisibilityPolicy.none);
+      await tester.tap(mute);
+      await tester.pump();
+      checkUpdateUserTopicRequest(UserTopicVisibilityPolicy.muted);
+    });
+
+    testWidgets('follow', (tester) async {
+      await setupToTopicActionSheet(tester,
+        isChannelMuted: false,
+        visibilityPolicy: UserTopicVisibilityPolicy.none);
+      await tester.tap(follow);
+      await tester.pump();
+      checkUpdateUserTopicRequest(UserTopicVisibilityPolicy.followed);
+    });
+
+    testWidgets('unfollow', (tester) async {
+      await setupToTopicActionSheet(tester,
+        isChannelMuted: false,
+        visibilityPolicy: UserTopicVisibilityPolicy.followed);
+      await tester.tap(unfollow);
+      await tester.pump();
+      checkUpdateUserTopicRequest(UserTopicVisibilityPolicy.none);
+    });
+
+    testWidgets('request fails with an error dialog', (tester) async {
+      await setupToTopicActionSheet(tester,
+        isChannelMuted: false,
+        visibilityPolicy: UserTopicVisibilityPolicy.followed);
+
+      connection.prepare(httpStatus: 400, json: {
+        'result': 'error', 'code': 'BAD_REQUEST', 'msg': ''});
+      await tester.tap(unfollow);
+      await tester.pumpAndSettle();
+
+      checkErrorDialog(tester, expectedTitle: 'Failed to unfollow topic');
+    });
+
+    group('check expected buttons', () {
+      final testCases = {
+        (false, UserTopicVisibilityPolicy.muted,    [unmute, follow]),
```

Comment:
nit: conceptually a list, not set
```suggestion
      final testCases = [
        (false, UserTopicVisibilityPolicy.muted,    [unmute, follow]),
```

(here and below)